src/internal/Subscriber.ts

```diff
@@ -230,7 +234,7 @@ export class SafeSubscriber<T> extends Subscriber<T> {
     if (!this.isStopped) {
       const { _parentSubscriber } = this;
       if (this._complete) {
-        const wrappedComplete = () => this._complete.call(this._context);
+        const wrappedComplete = () => this._complete.call(this);
```

Comment:
suggestion: This `wrappedComplete` business is not necessary. The `_complete` will have already been bound to the callers context - if an object was passed. `_complete` can be called directly.

Reply:
I am about to make a pass through all of this. I'm going to merge this as-is, and address the rest in another PR.